### PR TITLE
 Add isKeyboardShown command for iOS

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
## Change list

Add isKeyboardShown command for iOS

## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/HasDeviceDetails.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.appium.java_client;
+
+import static io.appium.java_client.MobileCommand.isKeyboardShownCommand;
+
+public interface HasDeviceDetails extends ExecutesMethod {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/HasDeviceDetails.java]: I'd rather rename the interface to HasOnScreenKeyboard

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 2]
```diff
@@ -64,7 +64,7 @@
 */
 @SuppressWarnings("unchecked")
 public class AppiumDriver<T extends WebElement>
-    extends DefaultGenericMobileDriver<T> implements ComparesImages {
+    extends DefaultGenericMobileDriver<T> implements ComparesImages, HasDeviceDetails {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: I'd rather add it to only ios  and Android drivers, but not to the generic one.

File: src/main/java/io/appium/java_client/HasDeviceDetails.java (hunk 1 above)
Comment (SrinivasanTarget) [reply src/main/java/io/appium/java_client/HasDeviceDetails.java]: Hmm makes sense

File: src/main/java/io/appium/java_client/android/AndroidDriver.java [hunk 3]
```diff
@@ -51,7 +51,7 @@
 public class AndroidDriver<T extends WebElement>
     extends AppiumDriver<T>
     implements PressesKeyCode, HasNetworkConnection, PushesFiles, StartsActivity,
-        FindsByAndroidUIAutomator<T>, LocksDevice, HasAndroidSettings, HasDeviceDetails,
+        FindsByAndroidUIAutomator<T>, LocksDevice, HasAndroidSettings, HasAndroidDeviceDetails,
         HasSupportedPerformanceDataType, AuthenticatesByFinger,
         CanRecordScreen, SupportsSpecialEmulatorCommands,
         SupportsNetworkStateManagement, ListensToLogcatMessages, HasAndroidClipboard {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/AndroidDriver.java]: The interface should be added to AndroidDriver as well

**Pull request merged**